api/src/main/java/org/openmrs/Cohort.java

```diff
@@ -213,8 +213,7 @@ public boolean removeMembership(CohortMembership cohortMembership) {
 	 * @since 2.1.0
 	 */
 	public CohortMembership getActiveMembership(Patient patient) {
-		return getMemberships().stream().filter(m -> m.isActive() && m.getPatientId().equals(patient.getPatientId()))
-		        .collect(Collectors.toList()).get(0);
+		return getMemberships().stream().filter(CohortMembership::isActive).filter(m -> m.getPatientId().equals(patient.getPatientId())).findFirst().get();
```

Comment:
Is there any reason why we are calling filter twice?

Reply:
Hi, I was given some feedback on the Google Code In task page suggesting that I use function pointers for the filter predicate. Seeing how this is inefficient I will revert the code back to having just one filter.